### PR TITLE
changed way of matching groups in LDAP for users

### LINKED ISSUE #1604: To be able to assign users to groups using LIKE % instead of complete name
Hello
I propose to give more possibilities when searching LDAP for user members of groups.
Let's have an example:
image
Current test is to match exact string in the attribute value. This has the drawback if 'department' is not written the exact same way in LDAP, then it will not match

I propose to use LIKE instead of IN in the mySQL query, This will permit to have a more flexible use of assignment:
image
Using LIKE operator will permit such criterion. In this case any user department matching the '163387 %' will be added to the group.

### PR DESCRIPTION
**Pull request by tomolimo**
Fixes #1604

### CODE REVIEW

File: inc/user.class.php [hunk 1]
```diff
@@ -1293,12 +1293,13 @@ private function getFromLDAPGroupVirtual($ldap_connection, $ldap_method, $userdn
                    && ($v[$i][$field]['count'] > 0)) {
 
                   unset($v[$i][$field]['count']);
+                  foreach( Toolbox::addslashes_deep($v[$i][$field]) as $lgroup){
```
Comment (trasher) [inline inc/user.class.php]: CS checks will break when cherry-picking into master ;)

Comment (tomolimo) [inline inc/user.class.php]: which one?

Comment (trasher) [inline inc/user.class.php]: s/one/ones/ ;)

```diff
diff --git a/inc/user.class.php b/inc/user.class.php
index ad6e183..f114fef 100644
--- a/inc/user.class.php
+++ b/inc/user.class.php
@@ -1293,8 +1293,8 @@ class User extends CommonDBTM {
                    && ($v[$i][$field]['count'] > 0)) {
 
                   unset($v[$i][$field]['count']);
-                  foreach( Toolbox::addslashes_deep($v[$i][$field]) as $lgroup){
-                     $lgroups[] = "('".$lgroup."' LIKE `ldap_value`)" ;
+                  foreach (Toolbox::addslashes_deep($v[$i][$field]) as $lgroup) {
+                     $lgroups[] = "('".$lgroup."' LIKE `ldap_value`)";
                   }
                   $query = "SELECT `id`
                             FROM `glpi_groups`
```

Comment (tomolimo) [inline inc/user.class.php]: Thank you, I understand now :), I'm going to fix this

**Comment by trasher** [comment]
LGTM. Waiting fo @wawax review since I'm not confident with ldap :)

**Pull request merged**